app_dart/lib/src/request_handlers/refresh_github_commits.dart

```diff
@@ -92,12 +90,17 @@ class RefreshGithubCommits extends ApiRequestHandler<Body> {
     }
 
     log.debug('Found ${newCommits.length} new commits on GitHub');
-    for (Commit commit in newCommits) {
+    /// [newCommit] has commits from latest to oldest
+    /// However, oldest is expected to be inserted first into datastore. Thus we 
+    /// need to reverse the order. At the same time, we update [timestamp]
+    for (Commit commit in newCommits.reversed.toList()) {
       final List<Task> tasks = await _createTasks(
         commitKey: commit.key,
         sha: commit.sha,
-        createTimestamp: now,
+        createTimestamp: DateTime.now().millisecondsSinceEpoch,
```

Comment:
Can we just take this from `commit.committer.date`?  That should give us the actual timestamp of when the commit was committed.

Reply:
commit.committer has two date related variables:
-----------
  /// The time this [User] was created.
  @JsonKey(name: "created_at")
  DateTime createdAt;

  /// Last time this [User] was updated.
  @JsonKey(name: "updated_at")
  DateTime updatedAt;
------------
They seem to be user's profile info., rather than commit time. Or you refer to something else?

Here we kind of care more about their relative time, rather than the exact commit time. Also, these (if more than one) should be within one minute (the cronjob period), do you think we can just use the now()?

Reply:
The actual JSON is:

```
{
    "sha": "71497367efc41eff684d8f092d467634f0875400",
    "node_id": "MDY6Q29tbWl0MzE3OTI4MjQ6NzE0OTczNjdlZmM0MWVmZjY4NGQ4ZjA5MmQ0Njc2MzRmMDg3NTQwMA==",
    "commit": {
      "author": {
        "name": "engine-flutter-autoroll",
        "email": "engine-flutter-autoroll@skia.org",
        "date": "2019-10-04T20:13:43Z"
      },
      "committer": {
        "name": "GitHub",
        "email": "noreply@github.com",
        "date": "2019-10-04T20:13:43Z"
      },
      "message": "Roll engine 759d3dba259b..7d90779bb66f (5 commits) (#42004)\n\n\ngit@github.com:flutter/engine.git/compare/759d3dba259b...7d90779bb66f\n\ngit log 759d3dba259b..7d90779bb66f --no-merges --oneline\n2019-10-04 skia-flutter-autoroll@skia.org Roll fuchsia/sdk/core/linux-amd64 from VkAgX... to by2LL... (flutter/engine#12791)\n2019-10-04 bkonyi@google.com Roll src/third_party/dart 28f21d4e99..d6c6d12ebf (3 commits)\n2019-10-04 jonahwilliams@google.com Fix bug in package script and add dev_compiler to list of uploaded packages (flutter/engine#12785)\n2019-10-04 bkonyi@google.com Roll src/third_party/dart 7a7dcd1ed9..28f21d4e99 (2 commits)\n2019-10-04 skia-flutter-autoroll@skia.org Roll fuchsia/sdk/core/mac-amd64 from wYLiQ... to JyZWz... (flutter/engine#12788)\n\n\nIf this roll has caused a breakage, revert this CL and stop the roller\nusing the controls here:\nhttps://autoroll.skia.org/r/flutter-engine-flutter-autoroll\nPlease CC liyuqian@google.com on the revert to ensure that a human\nis aware of the problem.\n\nTo report a problem with the AutoRoller itself, please file a bug:\nhttps://bugs.chromium.org/p/skia/issues/entry?template=Autoroller+Bug\n\nDocumentation for the AutoRoller is here:\nhttps://skia.googlesource.com/buildbot/+/master/autoroll/README.md",
      "tree": {
        "sha": "5e406e7b2e865f4f20f1d530dc86e30724a73a41",
        "url": "https://api.github.com/repos/flutter/flutter/git/trees/5e406e7b2e865f4f20f1d530dc86e30724a73a41"
      },
      "url": "https://api.github.com/repos/flutter/flutter/git/commits/71497367efc41eff684d8f092d467634f0875400",
      "comment_count": 0,
      "verification": {
        "verified": true,
        "reason": "valid",
        "signature": "-----BEGIN PGP SIGNATURE-----\n\nwsBcBAABCAAQBQJdl6f3CRBK7hj4Ov3rIwAAdHIIAFLyr0SV3s5zSNfeG+Hr54wM\nbichcygeNa+Kl86+rdtMnJW4zJgBek/GFTEsImhGO1XAD9EcaH7Bi6Jicu8Dg2/9\nxJro5yKv5QxoUmYZyGAtM4F6l1U9XkQabRvs83V+nTQ6QwqLtNOcVB9FNS9E7tA6\noO8G2kVBS3ygfHK3QObsdGZa/ARVGSz13gsYXyxqqURW6pd/lJ9ekC0hOJ+kv0y4\nUZ3Xv2ifgDDtWGhHXXHD2J1DCjRmZYiqKP8870GpEImlHFcWfNlgHyAT1BW0jYgY\nPa0/i+nEKEOh/M8AwHf4oXgPLzaG8w97/uIfUtVTCqu17cD2ZjKpj2m8nYBhTL4=\n=1xkw\n-----END PGP SIGNATURE-----\n",
        "payload": "tree 5e406e7b2e865f4f20f1d530dc86e30724a73a41\nparent 14c5c6e161dd720d1c1708e5b7b8f61cb4f85291\nauthor engine-flutter-autoroll <engine-flutter-autoroll@skia.org> 1570220023 -0400\ncommitter GitHub <noreply@github.com> 1570220023 -0400\n\nRoll engine 759d3dba259b..7d90779bb66f (5 commits) (#42004)\n\n\ngit@github.com:flutter/engine.git/compare/759d3dba259b...7d90779bb66f\n\ngit log 759d3dba259b..7d90779bb66f --no-merges --oneline\n2019-10-04 skia-flutter-autoroll@skia.org Roll fuchsia/sdk/core/linux-amd64 from VkAgX... to by2LL... (flutter/engine#12791)\n2019-10-04 bkonyi@google.com Roll src/third_party/dart 28f21d4e99..d6c6d12ebf (3 commits)\n2019-10-04 jonahwilliams@google.com Fix bug in package script and add dev_compiler to list of uploaded packages (flutter/engine#12785)\n2019-10-04 bkonyi@google.com Roll src/third_party/dart 7a7dcd1ed9..28f21d4e99 (2 commits)\n2019-10-04 skia-flutter-autoroll@skia.org Roll fuchsia/sdk/core/mac-amd64 from wYLiQ... to JyZWz... (flutter/engine#12788)\n\n\nIf this roll has caused a breakage, revert this CL and stop the roller\nusing the controls here:\nhttps://autoroll.skia.org/r/flutter-engine-flutter-autoroll\nPlease CC liyuqian@google.com on the revert to ensure that a human\nis aware of the problem.\n\nTo report a problem with the AutoRoller itself, please file a bug:\nhttps://bugs.chromium.org/p/skia/issues/entry?template=Autoroller+Bug\n\nDocumentation for the AutoRoller is here:\nhttps://skia.googlesource.com/buildbot/+/master/autoroll/README.md\n\n"
      }
    },
    "url": "https://api.github.com/repos/flutter/flutter/commits/71497367efc41eff684d8f092d467634f0875400",
    "html_url": "https://github.com/flutter/flutter/commit/71497367efc41eff684d8f092d467634f0875400",
    "comments_url": "https://api.github.com/repos/flutter/flutter/commits/71497367efc41eff684d8f092d467634f0875400/comments",
    "author": {
      "login": "engine-flutter-autoroll",
      "id": 42042535,
      "node_id": "MDQ6VXNlcjQyMDQyNTM1",
      "avatar_url": "https://avatars2.githubusercontent.com/u/42042535?v=4",
      "gravatar_id": "",
      "url": "https://api.github.com/users/engine-flutter-autoroll",
      "html_url": "https://github.com/engine-flutter-autoroll",
      "followers_url": "https://api.github.com/users/engine-flutter-autoroll/followers",
      "following_url": "https://api.github.com/users/engine-flutter-autoroll/following{/other_user}",
      "gists_url": "https://api.github.com/users/engine-flutter-autoroll/gists{/gist_id}",
      "starred_url": "https://api.github.com/users/engine-flutter-autoroll/starred{/owner}{/repo}",
      "subscriptions_url": "https://api.github.com/users/engine-flutter-autoroll/subscriptions",
      "organizations_url": "https://api.github.com/users/engine-flutter-autoroll/orgs",
      "repos_url": "https://api.github.com/users/engine-flutter-autoroll/repos",
      "events_url": "https://api.github.com/users/engine-flutter-autoroll/events{/privacy}",
      "received_events_url": "https://api.github.com/users/engine-flutter-autoroll/received_events",
      "type": "User",
      "site_admin": false
    },
    "committer": {
      "login": "web-flow",
      "id": 19864447,
      "node_id": "MDQ6VXNlcjE5ODY0NDQ3",
      "avatar_url": "https://avatars3.githubusercontent.com/u/19864447?v=4",
      "gravatar_id": "",
      "url": "https://api.github.com/users/web-flow",
      "html_url": "https://github.com/web-flow",
      "followers_url": "https://api.github.com/users/web-flow/followers",
      "following_url": "https://api.github.com/users/web-flow/following{/other_user}",
      "gists_url": "https://api.github.com/users/web-flow/gists{/gist_id}",
      "starred_url": "https://api.github.com/users/web-flow/starred{/owner}{/repo}",
      "subscriptions_url": "https://api.github.com/users/web-flow/subscriptions",
      "organizations_url": "https://api.github.com/users/web-flow/orgs",
      "repos_url": "https://api.github.com/users/web-flow/repos",
      "events_url": "https://api.github.com/users/web-flow/events{/privacy}",
      "received_events_url": "https://api.github.com/users/web-flow/received_events",
      "type": "User",
      "site_admin": false
    },
    "parents": [
      {
        "sha": "14c5c6e161dd720d1c1708e5b7b8f61cb4f85291",
        "url": "https://api.github.com/repos/flutter/flutter/commits/14c5c6e161dd720d1c1708e5b7b8f61cb4f85291",
        "html_url": "https://github.com/flutter/flutter/commit/14c5c6e161dd720d1c1708e5b7b8f61cb4f85291"
      }
    ]
  },
```

We want `commit.committer.date`.

Reply:
Updated as suggested. With this committer date, we do not need to consider reversing order when inserting data to datastore. Thank you Dan!